packages/protocol/engine/congestioncontrol/icca/scheduler/scheduler.go

```diff
@@ -233,6 +230,14 @@ func (s *Scheduler) AddBlock(sourceBlock *virtualvoting.Block) {
 
 	block, _ := s.getOrRegisterBlock(sourceBlock)
 
+	if block.IsOrphaned() {
+		if !block.IsDropped() {
```

Comment:
if you returned an `updated` from `SetDropped` you could inline this here

Reply:
👍